src/Model/Game.java

```diff
@@ -60,20 +61,41 @@ public PlayerInfo findPlayerInfo(String user){
         return playerInfo;
     }
 
+    public ArrayList<iCard> replaceAllFaceUp(){
+        int locomotiveCounter = 0;
+        ArrayList<iCard> newHand = new ArrayList<>();
+        for (int i = 0; i < 5; i++){
+            iCard card = gameTrainDeck.draw();
+            newHand.add(card);
+            if (((TrainCarCard)card).getType() == TrainCarCardType.LOCOMOTIVE){
+                locomotiveCounter += 1;
+            }
+        }
+
+        if (locomotiveCounter >= 3){
+            replaceAllFaceUp();
```

Comment:
I don't think this recursion would work. You're not using the return statement from replaceAllFaceUp(). If the new 5 cards had 3 or more locomotives, then you would call the function again, but return the bad hand a few lines later. Replacing this with "return replaceAllFaceUp();" might fix it, but I would test it to make sure.